source/extensions/filters/http/rbac/rbac_filter.cc

```diff
@@ -80,24 +80,19 @@ Http::FilterHeadersStatus RoleBasedAccessControlFilter::decodeHeaders(Http::Head
       shadow_resp_code = resp_code_403;
     }
 
-    const auto& filter_metadata = callbacks_->requestInfo().dynamicMetadata().filter_metadata();
-    const auto filter_it = filter_metadata.find(HttpFilterNames::get().Rbac);
-    if (filter_it != filter_metadata.end()) {
-      ProtobufWkt::Struct metrics;
-
-      if (!effective_policy_id.empty()) {
-        ProtobufWkt::Value policy_id;
-        policy_id.set_string_value(effective_policy_id);
-        (*metrics.mutable_fields())[shadow_policy_id_field] = policy_id;
-      }
-
-      ProtobufWkt::Value resp_code;
-      resp_code.set_string_value(shadow_resp_code);
-      (*metrics.mutable_fields())[shadow_resp_code_field] = resp_code;
-
-      auto filter_meta = filter_metadata.at(HttpFilterNames::get().Rbac);
-      filter_meta.MergeFrom(metrics);
+    ProtobufWkt::Struct metrics;
+
+    if (!effective_policy_id.empty()) {
+      ProtobufWkt::Value policy_id;
+      policy_id.set_string_value(effective_policy_id);
+      (*metrics.mutable_fields())[shadow_policy_id_field] = policy_id;
```

Comment:
It would be nice if you can avoid one of copies involved here, something like:

```
auto& fields = *metrics.mutable_fields();
*fields[shadow_policy_id_field].mutable_string_value() = policy_id;
```

same for resp_code below.

Reply:
thanks, updated.